sheetstorm/system_check.py

```diff
@@ -0,0 +1,123 @@
+from django.core.checks import register
+
+from common.constants import PROHIBITED_SIGNS_IN_VALID_EMAIL_DOMAIN_LIST
+
+
+def create_error_01_valid_email_domain_list_is_not_a_list() -> Error:
+    return Error("The value of VALID_EMAIL_DOMAIN_LIST setting is not a list", id="sheetstorm.E001")
+
+
+def create_warning_01_duplicated_values_in_valid_email_domain_list() -> Warning:
+    return Warning(
+        "Some domains in VALID_EMAIL_DOMAIN_LIST are duplicated", hint="Remove duplicate domains", id="sheetstorm.W001"
+    )
+
+
+def create_error_02_invalid_domain_address(domain: str) -> Error:
+    return Error(
+        "Domain {} doesn't have at least one dot".format(domain),
+        hint="Add valid domain address for instance - 'instance.com'",
+        id="sheetstorm.E002",
+    )
+
+
+def create_error_03_domain_contains_prohibited_signs(domain: str) -> Error:
+    return Error(
+        "Domain {} contains prohibited signs".format(domain),
+        hint="Remove prohibited signs from domain address",
+        id="sheetstorm.E003",
+    )
+
+
+def create_warning_02_empty_valid_email_domain_list() -> Warning:
+    return Warning(
+        "VALID_EMAIL_DOMAIN_LIST doesn't contain any domains, all domains are invalid now",
+        hint="Add valid domains to VALID_EMAIL_DOMAIN_LIST",
+        id="sheetstorm.W002",
+    )
+
+
+def create_error_04_site_id_value_must_be_integer() -> Error:
+    return Error("SITE_ID value must be an integer", id="sheetstorm.E004")
+
+
+def create_error_05_email_backend_value_must_be_string() -> Error:
+    return Error("EMAIL_BACKED value must be a string", id="sheetstorm.E005")
+
+
+def create_error_06_email_backend_library_does_not_exist() -> Error:
+    return Error(f"File doesn't exist", id="sheetstorm.E006")
+
+
+def create_error_07_email_backend_file_does_not_have_attribute() -> Error:
+    return Error(f"Attribute from module {settings.EMAIL_BACKEND} doesn't exist", id="sheetstorm.E007")
+
+
+def create_08_email_backend_one_of_modules_does_not_exist() -> Error:
+    return Error(f"Module from path: {settings.EMAIL_BACKEND} can not be imported", id="sheetstorm.E008")
+
+
+@register()
+def check_settings_valid_email_domain_list(
+    app_configs: None = None, **kwargs: Any  # pylint: disable=unused-argument
+) -> list:
+    if not isinstance(settings.VALID_EMAIL_DOMAIN_LIST, list):
+        return [create_error_01_valid_email_domain_list_is_not_a_list()]
+
+    if len(settings.VALID_EMAIL_DOMAIN_LIST) == 0:
+        return [create_warning_02_empty_valid_email_domain_list()]
+
+    errors_and_warnings = []
+    if len(set(settings.VALID_EMAIL_DOMAIN_LIST)) != len(settings.VALID_EMAIL_DOMAIN_LIST):
+        errors_and_warnings.append(create_warning_01_duplicated_values_in_valid_email_domain_list())
+
+    for domain in settings.VALID_EMAIL_DOMAIN_LIST:
+        if "." not in domain:
+            errors_and_warnings.append(create_error_02_invalid_domain_address(domain))
+        contains_prohibited_signs = any(
```

Comment:
How about using regex for this? Here is the example https://gist.github.com/jpadilla/8659759.

Reply:
I could use regex but it wouldn't be hard to understand what is going on there ?

Reply:
Done

Reply:
Does it work with `localhost`? :)

Reply:
I just change it to regex. As I wrote before on chat I am waiting for your opinion what to do with `localhost`

Reply:
OK, please make it work for `localhost` too, thanks.